app/models/visualization.rb

```diff
@@ -1,5 +1,9 @@
 class Visualization < ApplicationRecord
-  has_many :variables, inverse_of: :visualization
+  has_many :filters, :variables, inverse_of: :visualization
+  validates :chart_title, presence: true
+  validates :x_axis_title, presence: true
+  validates :y_axis_title, presence: true
+  validates :chart_type, presence: true
   has_many :filters, inverse_of: :visualization
```

Comment:
Why is this line repeated?

Reply:
I think we either just moved it or somebody fixed it.  Thanks!